### PR TITLE
Don't do iso14443-4 select for mifare emulations on processor cards (IDPrime, SmartMX, …)

### PR DESCRIPTION
**Pull request by pwpiwi**
fixing issue #406

### CODE REVIEW

**Comment by iceman1001** [comment]
So, RATS is needed for all kinds of Mifare tags?  All Classic, UL,UL-C  or?

**Comment by iceman1001** [comment]
and RATS is needed for Topaz (cmdhftopaz.c) ?

**Comment by pwpiwi** [comment]
RATS is ISO14443-4. All Mifare are 14443-3 only.

**Comment by iceman1001** [comment]
Good, and Topaz is?

https://github.com/Proxmark/proxmark3/blob/master/client/cmdhftopaz.c#L64

**Comment by pwpiwi** [comment]
I don't know about a topaz emulation in processor cards

**Comment by iceman1001** [comment]
Good, in the datasheet I have there is only references to 14443-3,  not 4.  So NO_RATS for Topaz.  You will to add that to this PR.

**Comment by iceman1001** [comment]
Not to mention the LUA scripts will also need to be adapted in your PR.

**Comment by pwpiwi** [comment]
Re Topaz: it doesn't support a select command at all. It therefore doesn't matter if ISO14A_NO_RATS is set or not.

File: client/scripts/14araw.lua [hunk 1]
```diff
@@ -83,23 +86,27 @@ function main(args)
 	local stayconnected = false
 	local payload = nil
 	local doconnect = true
+	local topaz_mode = false
+	local no_rats = false
 
 	-- Read the parameters
-	for o, a in getopt.getopt(args, 'corcpx:') do
+	for o, a in getopt.getopt(args, 'corcpxt3:') do
```
Comment (iceman1001) [inline client/scripts/14araw.lua]: You got it wrong.  
The : indicates that the param char has more bytes to read.
It should be like before,  x:  meaning payload bytes after x,   no 3:

File: client/scripts/14araw.lua [hunk 2]
```diff
@@ -137,11 +144,11 @@ function showdata(usbpacket)
 end
 
 
-
 function sendRaw(rawdata, options)
 	print(">> ", rawdata)
 	
 	local flags = lib14a.ISO14A_COMMAND.ISO14A_NO_DISCONNECT + lib14a.ISO14A_COMMAND.ISO14A_RAW
+	if topaz_mode == true then flags = flags + lib14a.ISO14A_COMMAND.ISO14A_TOPAZMODE end
```
Comment (iceman1001) [inline client/scripts/14araw.lua]: the variable topaz_mode is not global.  You need to call it like  options is done.

Comment (pwpiwi) [reply client/scripts/14araw.lua]: You might have guessed already: I don't speak LUA. 

So it should be `'corcpx:t3'` ? And btw: why do we have `c` two times?

So it should be `options.topaz_mode` instead?

Comment (iceman1001) [reply client/scripts/14araw.lua]: The double c is redundant aswell.  One is enough.

The options param, needs to be called with topaz_mode ( https://github.com/iceman1001/proxmark3/blob/master/client/scripts/14araw.lua#L111 )

and use/checked like you wrote.

Comment (iceman1001) [reply client/scripts/14araw.lua]: looking at the getopt,  I'm guessing it should be a 'd'  in order to enable debug mode, instead of an extra 'c'

**Pull request merged**